### PR TITLE
Added SetVariableFromXML task that creates Whiskey variables from values XML files.

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Test/SetVariableFromXml.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    Path: fubar.xml
+    Variables:
+        Var1: /root/element/fubar
+'@
+    WhenRunningTask
+    ThenVariable 'Var1' -Is ''
+    ThenNoErrors
+}
+
+Describe 'SetVariableFromXml.when XML has namespaces' {
```
Comment (jlariono) [inline Test/SetVariableFromXml.Tests.ps1]: Maybe add a test for **multiple** namespaces other than just a default one?

File: Whiskey/en-US/about_Whiskey_SetVariableFromXml.help.txt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    If an element/attribute doesn't exist at a given XPath, the variable's value will be set to an empty string.
+
+    If an XML document is namespaced (i.e. its root element has an "xmlns" attribute), you must use the "NamespacePrefixes" property to give the namespace a prefix and your XPath expression must prefix each element in that namespace with that prefix. See "Example 3" for sample code.
+
+PROPERTIES
+    * `Path` (***mandatory***): the path to the XML file to read.
+    * `Variables` (***mandatory***): an object whose properties are the variable names. The value of each variable is the XPath expression to use to select the variable's value.
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Whiskey/en-US/about_Whiskey_SetVariableFromXml.help.txt]: YAML docs use the word "mappings". I've updated the docs to use this word.

File: Whiskey/Tasks/Set-WhiskeyVariableFromXml.ps1 [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        $TaskContext,
+
+        [Parameter(Mandatory=$true)]
+        [hashtable]
+        $TaskParameter
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+    $VerbosePreference = 'Continue'
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Whiskey/Tasks/Set-WhiskeyVariableFromXml.ps1]: Fixed.

File: Whiskey/Tasks/Set-WhiskeyVariableFromXml.ps1 [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    
+    $path = $TaskParameter['Path'] | Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'Path' -PathType File
+    if( ($path | Measure-Object).Count -ne 1 )
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -PropertyName 'Path' -Message ('resolves to multiple files. The "Path" property must only resolve to one file.')
+    }
+
+    Write-WhiskeyVerbose -Context $TaskContext -Message ($path)
+    [xml]$xml = Get-Content -Path $path -Raw
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Whiskey/Tasks/Set-WhiskeyVariableFromXml.ps1]: Fixed.

File: Test/SetVariableFromXml.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,232 @@
+
+#Requires -Version 4
+Set-StrictMode -Version 'Latest'
+
+& (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-WhiskeyTest.ps1' -Resolve)
+
+$path = $null
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Test/SetVariableFromXml.Tests.ps1]: Removed.

File: Test/SetVariableFromXml.Tests.ps1 (hunk 1 above)
Comment (splatteredbits) [reply Test/SetVariableFromXml.Tests.ps1]: Added.

File: Whiskey/en-US/about_Whiskey_SetVariableFromXml.help.txt [hunk 6]
```diff
@@ -0,0 +1,51 @@
+TOPIC
+    about_Whiskey_SetVariableFromXml_Task
+
+SUMMARY
+    Sets Whiskey variables by reading values from XML files.
+
+DESCRIPTION
+    The `SetVariableFromXml` task sets Whiskey variables by reading values from XML files. The path to the XML file to read is set with the `Path` property. Variable values are selected using XPath expressions. The names of the variables along with their XPath expressions are set with the `Variables` property. Every property under `Variables` is the name of the variable to set. Its value is the XPath expression to use. The inner text of the selected XML element or attribute is used as the value of the variable. If the XPath expression matches multiple elements/attributes, the value of the variable will be a list/array of inner text values.
+
+    If an element/attribute doesn't exist at a given XPath, the variable's value will be set to an empty string.
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Whiskey/en-US/about_Whiskey_SetVariableFromXml.help.txt]: Changed default behavior to fail if a node is missing. Added `AllowMissingNodes` property to allow users to let XPath expressions not return any results.

**Comment by splatteredbits** [comment]
@jlariono Ready for re-review.

File: Test/SetVariableFromXml.Tests.ps1 [hunk 7]
```diff
@@ -31,6 +30,18 @@ function Init
 {
     $script:path = $null
```
Comment (jlariono) [inline Test/SetVariableFromXml.Tests.ps1]: This variable was removed.

**Pull request merged**